pkg/cmd/node.go

```diff
@@ -0,0 +1,84 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	"github.com/spf13/cobra"
+
+	"github.com/kubecost/kubectl-cost/pkg/query"
+)
+
+// CostOptionsNode contains the standard CostOptions and any
+// options specific to node queries.
+type CostOptionsNode struct {
+	CostOptions
+}
+
+func newCmdCostNode(streams genericclioptions.IOStreams) *cobra.Command {
+	kubeO := NewKubeOptions(streams)
+	assetsO := &CostOptionsNode{}
+
+	cmd := &cobra.Command{
+		Use:   "node",
+		Short: "view cost information by nodes",
+		RunE: func(c *cobra.Command, args []string) error {
+			if err := kubeO.Complete(c, args); err != nil {
+				return err
+			}
+			if err := kubeO.Validate(); err != nil {
+				return err
+			}
+
+			assetsO.CostOptions.Complete()
+
+			if err := assetsO.CostOptions.Validate(); err != nil {
+				return err
+			}
+
+			return runCostNode(kubeO, assetsO)
+		},
+	}
+
+	addCostOptionsFlags(cmd, &assetsO.CostOptions)
+	addKubeOptionsFlags(cmd, kubeO)
+
+	return cmd
+}
+
+func runCostNode(ko *KubeOptions, no *CostOptionsNode) error {
+
+	currencyCode, err := query.QueryCurrencyCode(query.CurrencyCodeParameters{
+		RestConfig:        ko.restConfig,
+		Ctx:               context.Background(),
+		KubecostNamespace: *ko.configFlags.Namespace,
+		ServiceName:       no.serviceName,
+		UseProxy:          no.useProxy,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to get currency code: %s", err)
+	}
+
+	assets, err := query.QueryAssets(query.AssetParameters{
+		RestConfig:         ko.restConfig,
+		Ctx:                context.Background(),
+		KubecostNamespace:  *ko.configFlags.Namespace,
+		ServiceName:        no.serviceName,
+		Window:             no.window,
+		Aggregate:          "",
+		DisableAdjustments: false,
```

Comment:
Go has an interesting concept of "zero values" (https://tour.golang.org/basics/12). If `Aggregate` is of type string, then if you don't set it when you make a new `query.AssetParameters` struct it will default to `""`. Likewise for `DisableAdjustments` and `false`.

Reply:
Yes, honestly I just kept them there to remind me of the default values. I could delete them if you feel that's better/cleaner.